crates/modules/process-monitor/probes.bpf.c

```diff
@@ -96,19 +124,76 @@ struct pending_dead_process {
 // Temporary map used to communicate the list of orphaned processes from
 // `sched_process_exit` to `sched_switch`. There we'll be able to read
 // the new parent of the orphans and emit an EVENT_CHANGE_PARENT.
-struct {
+struct
+{
   __uint(type, BPF_MAP_TYPE_QUEUE);
   __type(value, struct pending_dead_process);
   __uint(max_entries, MAX_PENDING_DEAD_PARENTS);
 } orphans_map SEC(".maps");
 
-struct {
+struct
+{
   __uint(type, BPF_MAP_TYPE_HASH);
   __type(key, u64);
   __type(value, u8);
   __uint(max_entries, MAX_CONTAINER_RUNTIMES);
 } container_runtimes_map SEC(".maps");
 
+static __always_inline int get_cgroup_name(char *buf, size_t sz, int *offset, struct task_struct *cur_tsk)
+{
+
+  // int cgrp_id = memory_cgrp_id;
+  // int cgrp_id = cpu_cgrp_id;
+  int cgrp_id = 4;
+
+  // failed when use BPF_PROBE_READ
+  const char *name = BPF_CORE_READ(cur_tsk, cgroups, subsys[cgrp_id], cgroup, kn, name);
+  // bpf_printk("name: %s\n", name);
+  if (bpf_probe_read_kernel_str(buf, sz, name) < 0)
+  {
+    bpf_printk("failed to get kernfs node name: %s\n", buf);
+    return -2;
+  }
+  pid_t tgid = BPF_CORE_READ(cur_tsk, tgid);
+
+  // Docker case
+  if (bpf_strncmp(buf, 7, "docker-") == 0)
+  {
+    // bpf_printk("docker case: %s\n", buf);
+    *offset = 7;
+    return 0;
+  }
+
+  // Podman case
+  if (bpf_strncmp(buf, 9, "container") == 0 && buf[9] == '\0')
+  {
+
+    const char *parent_name = BPF_CORE_READ(cur_tsk, cgroups, subsys[cgrp_id], cgroup, kn, parent, name);
+    // bpf_printk("podman case step 1: parent name: %s\n", parent_name);
+
+    if (bpf_probe_read_kernel_str(buf, sz, parent_name) < 0)
+    {
+      bpf_printk("failed to get kernfs node name: %s\n", buf);
+      return -3;
```

Comment:
Could we `#define` the error codes with descriptive names?

Reply:
sure